interp/interpreter.go

```diff
@@ -24,15 +24,36 @@ func (r *runner) run(fn *function, params []value, parentMem *memoryView, indent
 		locals[i] = param
 	}
 
+	// Track what blocks have run instructions at runtime.
+	// This is used to prevent unrolling.
+	runtimeBlocks := make(map[int]struct{})
```

Comment:
I've noticed that heap allocations are a significant slowdown, so I think it would be a good idea to move the creation of the map to where anything is inserted. Like:

```suggestion
	var runtimeBlocks map[int]struct{}
```

And then:
```go
				// Flag the last block as having run stuff at runtime.
				if runtimeBlocks == nil {
					runtimeBlocks = make(map[int]struct{})
				}
				runtimeBlocks[lastBB] = struct{}{}
```

Reply:
In this case, the map will be used in any situation with multiple basic blocks. It might be more efficient for me to reuse these maps within an interp pass instead.

Reply:
Only when there are instructions run at runtime, right? That should not happen very often. Most functions should go through interp without any instructions emitted at runtime.

Reply:
Oh yeah, was just a bit confused coming back to this after a while.

Reply:
Done